### DISCUSSION TITLE
Keep losing session periodically

### DISCUSSION THREAD

**Discussion by enoryw** [jarun/nnn#1893] [Q&A]
I keep losing sessions periodically, how should I go about debugging this? I autostart a terminal started with tmux that runs multiple nnn instances each with their own sessions. I don't explicitly manually quit the sessions on system shutdown (I know nnn doesn't (can't?) save its sessions if nnn is terminated externally i.e. not manually quitting the sessions with a `Q`) but I already saved sessions manually and they are restored as you would except until eventually the session is lost/reset.  I also restart the tmux instance on occassion which "abruptly" restarts nnn instances (again, this is usually fine).

I do use `make O_CTX8=1 O_NERD=1 O_QSORT=1 O_NOX11=1` but I don't think that should increase additional risks of losing sessions so I'm curious if there are any known ways a session may be lost/corrupt.

**Comment by jarun** [comment]
`nnn` doesn't have logic to delete session files. They can be updated.

Are the session file deleted? If not, probably they are being overwritten by  difference instances of `nnn`.
Please check if multiple instances use the same session name.
